indico/modules/announcement/templates/display.html

```diff
@@ -1,23 +1,80 @@
+{% macro render_announcement_legacy(type, message, persistent=false) %}
+    {% if type == 'html' %}
+        {{ message | safe }}
+    {% else %}
+        <div class="announcement-bar weak-hidden {{type}}" data-hash={{message | crc32 | string | tojson}}>
+            <div class="flexrow f-self-stretch f-j-center f-a-center">
+                <i class="icon icon-warning"></i>
+                <div>
+                    {{ message }}
+                </div>
+            </div>
+            {% if not persistent %}
+                <span class="icon-close close-button"></span>
+            {% endif %}
+        </div>
+    {% endif %}
+{% endmacro %}
 
-        var hash = {{ message | crc32 | string | tojson }};
+{% macro render_announcement_new(type, message, persistent=false) %}
+    {% if type == 'html' %}
+        {{ message | safe }}
+    {% elif type == 'logged-in-as' %}
+    {% else %}
+        <div class="announcement-bar weak-hidden {{type}}" data-hash={{message | crc32 | string | tojson}}>
+            <div class="content">
+                <i class="warning sign large icon"></i>
+                <div class="message">
+                    {{ message }}
+                </div>
+                <i class="close icon close-button"></i>
+            </div>
+        </div>
+    {% endif %}
+{% endmacro %}
 
-        $('#announcement-bar .close-button').on('click', function() {
-            $.jStorage.set('hideAnnouncement', hash);
-            $('#announcement-bar').slideUp('fast');
-        });
+{% macro render_announcements(legacy=false) %}
+    {% set announcements = template_hook('global-announcement', as_list=true) %}
+    {% if announcements %}
+        {% for announcement in announcements %}
+            {% if legacy %}
+                {{ render_announcement_legacy(*announcement) }}
+            {% else %}
+                {{ render_announcement_new(*announcement) }}
+            {% endif %}
+        {% endfor %}
+        <script>
+            function refreshPageHeight() {
+                let numBars = Array.from(document.querySelectorAll('.announcement-bar'))
+                    .map(function(e) { return e.style.display === 'flex' })
+                    .filter(function(e) { return !!e; })
+                    .length;
 
-        if ($.jStorage.get('hideAnnouncement') !== hash) {
-            $("#announcement-bar").show();
-        }
-    })();
-</script>
+                if (document.querySelector('.logged-in-as-bar')) {
+                    numBars++;
+                }
+
+                document.querySelector('body').style.setProperty('--extra-bars', numBars);
+            }
+
+            (() => {
+                document.querySelectorAll('.announcement-bar .close-button').forEach(function(elem) {
+                    elem.addEventListener('click', function() {
+                        const bar = elem.closest('.announcement-bar');
+                        window.localStorage.setItem('hideAnnouncement', bar.dataset.hash);
```

Comment:
`window.` isn't needed since it's global anyway

Reply:
I know. But I never know whether I should use it or not.

Reply:
Actually, at a second glance, I tend to use `window.` when the global in question is a DOM/browser global, and not just a language global. So, I'd never write `window.setTimeout`, but `window.localStorage` looks logical to me.
Do you think we should always omit `window`?

Reply:
no strong opinion